master/pkg/stream/stream_test.go

```diff
@@ -36,10 +38,46 @@ func (tm TestMsg) DeleteMsg() DeleteMsg {
 	}
 }
 
+// A second test message type to help test that msgs are being distinguished from each other.
+type TestMsgJr struct {
+	Seq int64
+	ID  int
+}
+
+func (em TestMsgJr) SeqNum() int64 {
+	return em.Seq
+}
+
+func (em TestMsgJr) UpsertMsg() UpsertMsg {
+	return UpsertMsg{
+		JSONKey: TestMsgJrUpsertKey,
+		Msg:     em,
+	}
+}
+
+func (em TestMsgJr) DeleteMsg() DeleteMsg {
+	deleted := strconv.FormatInt(int64(em.ID), 10)
+	return DeleteMsg{
+		Key:     TestMsgJrDeleteKey,
+		Deleted: deleted,
+	}
+}
+
 func alwaysTrue(msg TestMsg) bool {
 	return true
 }
 
+func alwaysTrueJr(msg TestMsgJr) bool {
```

Comment:
Could an interface be used here? So that we don't have to duplicate code?

Reply:
good point! I don't know about an interface, but we could definitely make the function generic